Feature/add client repository

Maria-Zurakowska opened the pull request:
1. The code consisted of business entity classes as well as CourseRepository interface with an implementing class (ListBasedCourseOperator)
2. The project required additional repository - ClientRepository to handle client management
3. I've added a new interface - ClientRepository with typical CRUD methods to handle client management:
createClient(String name, String lastName), printAllClients(), findClientsByLastName(String lastName), updateClient(Client client, String name, String lastName) and deleteClient(Client client)
and their implementation | I've also added ClientOperator methods to School class for it to be responsible for governing Clients

Maria-Zurakowska committed:
Add draft methods for ClientRepository

Maria-Zurakowska committed:
Add Builder design pattern to Client class

Maria-Zurakowska committed:
Add implementation of createClient(String name, String lastName) method with testing in 'Runner' class

PaniOdProgramowania commented on src/main/java/com/koreanlanguageschool/KoreanLanguageSchoolApplication.java:
```diff
@@ -36,4 +35,13 @@ private static void getCourseClients() {
 		System.out.println(clientsByCourse);
 	}
+
+	private static void createClient(){
```
let's rename this method - the name `create` indicates something would be returned but it's actually `void`... so I think a better name would be even `test`. Or, maybe even put the inside of this method into the `main` method and test the `operator` there :)

Maria-Zurakowska replied:
Ok, done

PaniOdProgramowania reviewed (commented):
so far it looks good :) I added just one comment. Please apply changes and then let me know once this is ready for re-reviewing:)

Maria-Zurakowska committed:
Remove createClient method from 'Runner' class

Maria-Zurakowska committed:
Remove redundant class

Maria-Zurakowska committed:
Add implementation of printAllClients method with test in 'Runner' class

Maria-Zurakowska committed:
Add implementation of List<Client> findClientsByLastName(String lastName) method including scenario when user types last name in lower case

Maria-Zurakowska committed:
Add implementation of updateClient(Client client, String name, String lastName) with testing in 'Runner' class

Maria-Zurakowska committed:
Add implementation of deleteClient(Client client) method with test in 'Runner' class

Maria-Zurakowska committed:
Add ClientOperator methods to School class for it to be responsible for governing Clients

Maria-Zurakowska committed:
Add change of dateOfEmployment field type in Employee class from Instant to LocalDate
Add @AllArgsConstructor annotation to School class
Fix NullPointerException in 'Runner' class when invoking methods from School class

Maria-Zurakowska commented:
I needed to create a School object with all of its parameters in order to later invoke the printAllClients() method from the said class

Maria-Zurakowska committed:
Add Builder design pattern to School class in order to create objects more efficiently

PaniOdProgramowania reviewed (approved):
great! feel free to merge 🎖️

Maria-Zurakowska commented:
Ready to merge
